referenceframe/model_json.go

```diff
@@ -4,154 +4,85 @@ import (
 	"encoding/json"
 	"os"
 
-	"github.com/golang/geo/r3"
 	"github.com/pkg/errors"
-
-	spatial "go.viam.com/rdk/spatialmath"
-	"go.viam.com/rdk/utils"
 )
 
 // ModelConfig represents all supported fields in a kinematics JSON file.
 type ModelConfig struct {
-	Name         string           `json:"name"`
-	KinParamType string           `json:"kinematic_param_type"`
-	Links        []JSONLink       `json:"links"`
-	Joints       []JSONJoint      `json:"joints"`
-	DHParams     []JSONDHParam    `json:"dhParams"`
-	RawFrames    []FrameMapConfig `json:"frames"`
-}
-
-// JSONLink is a struct which details the JSON used in a ModelJSON link element.
-type JSONLink struct {
-	ID          string                    `json:"id"`
-	Parent      string                    `json:"parent"`
-	Translation spatial.TranslationConfig `json:"translation"`
-	Orientation spatial.OrientationConfig `json:"orientation"`
-	Geometry    spatial.GeometryConfig    `json:"geometry"`
-}
-
-// JSONJoint is a struct which details the JSON used in a ModelJSON joint element.
-type JSONJoint struct {
-	ID     string             `json:"id"`
-	Type   string             `json:"type"`
-	Parent string             `json:"parent"`
-	Axis   spatial.AxisConfig `json:"axis"`
-	Max    float64            `json:"max"` // in mm or degs
-	Min    float64            `json:"min"` // in mm or degs
-}
-
-// JSONDHParam is a struct which details the JSON used to describe DH parameters.
-type JSONDHParam struct {
-	ID       string                 `json:"id"`
-	Parent   string                 `json:"parent"`
-	A        float64                `json:"a"`
-	D        float64                `json:"d"`
-	Alpha    float64                `json:"alpha"`
-	Max      float64                `json:"max"` // in mm or degs
-	Min      float64                `json:"min"` // in mm or degs
-	Geometry spatial.GeometryConfig `json:"geometry"`
+	Name         string          `json:"name"`
+	KinParamType string          `json:"kinematic_param_type,omitempty"`
+	Links        []LinkConfig    `json:"links,omitempty"`
+	Joints       []JointConfig   `json:"joints,omitempty"`
+	DHParams     []DHParamConfig `json:"dhParams,omitempty"`
```

Comment:
Thank you for this, much cleaner now that we have common structs in a sensible location